pkg/porter/explain.go

```diff
@@ -328,6 +328,21 @@ func (p *Porter) printBundleExplainTable(bun *PrintableBundle, bundleReference s
 	p.printDependenciesExplainBlock(bun)
 
 	fmt.Fprintf(p.Out, "This bundle uses the following tools: %s.\n", strings.Join(bun.Mixins, ", "))
+
+	// Check whether the bundle requires docker mixin and warn for host access.
+	var requireDocker bool = false
+	for _, mixin := range bun.Mixins {
```

Comment:
Using the docker mixin isn't the problem we need to warn about. What we are looking for is when a bundle declares that it requires docker host access:

https://porter.sh/author-bundles/#required

```
required:
  - docker # this says inject the docker socket into the bundle
```

The preferred way to check for that is with `bun.SupportsX`, like `bun.SupportsFileParameters`, which checks if a CNAB required extension is defined on the bundle. There _should_ be a function named `bun.SupportsDocker` but I guess that function never made.

It should look like this:

https://github.com/getporter/porter/blob/6dab64a39100ff6cbf1aecbeaaf58b377f69a7fa/pkg/cnab/extensions/parameter_sources.go#L198

which should be added to https://github.com/getporter/porter/blob/main/pkg/cnab/extensions/docker.go

Reply:
I just realized that I didn't stumble upon this one since the directory of `pkg/cnab/extensions` doesn't exist on `release/v1` branch but only exists on `main` branch. Does this mean that the PR for this particular issue should be made upon the `main` branch?

Reply:
Oops, sorry I shouldn't have linked to the main branch. This PR should target release/v1 but the code that I wanted to show you as an example moved a bit: https://github.com/getporter/porter/blob/a93a44a1bbf1e43bf96a80310b34e33a90de52f2/pkg/cnab/parameter_sources.go#L204-L206.